mmdet3d/models/segmentors/encoder_decoder.py

```diff
@@ -417,20 +450,22 @@ def slide_inference(self, point: Tensor, img_meta: List[dict],
 
         return preds.transpose(0, 1)  # to [num_classes, K*N]
 
-    def whole_inference(self, points: Tensor, input_metas: List[dict],
+    def whole_inference(self, points: Tensor, batch_data_samples: SampleList,
                         rescale: bool) -> Tensor:
         """Inference with full scene (one forward pass without sliding)."""
-        seg_logit = self.encode_decode(points, input_metas)
+        seg_logit = self.encode_decode(points, batch_data_samples)
         # TODO: if rescale and voxelization segmentor
         return seg_logit
 
-    def inference(self, points: Tensor, input_metas: List[dict],
+    def inference(self, points: Tensor, batch_data_samples: SampleList,
```

Comment:
Are you sure this modification? This may have conflicts with line 479. I think `input_metas` is OK. We just keep `batch_data_sample` interface in  high-level functions like, `predict` and `loss`. Also, `inference` does not need annotation contained in DataSample.